test/plugin.test.js

```diff
@@ -140,6 +151,234 @@ QUnit.test('no progress for 45 seconds is an error', function(assert) {
+
+  this.player.on('error', function() {
+    errors++;
+  });
+  this.player.src(sources);
+
+  this.player.trigger('play');
+
+  this.clock.tick(1 * 1000);
+
+  assert.ok(
+    this.player.hasClass('vjs-waiting'),
+    'the plugin adds spinner class to the player after 1 sec of no progress'
+  );
+
+  // document becomes hidden
+  document.visibilityState = 'hidden';
+  Events.trigger(document, 'visibilitychange');
+
+  this.clock.tick(9 * 1000);
+
+  assert.strictEqual(errors, 0, 'did not emit an error in background after 9 sec');
+
+  this.clock.tick(1 * 1000);
+
+  assert.strictEqual(errors, 1, 'emitted an error after 10 sec');
+  assert.strictEqual(this.player.error().code, -2, 'error code is -2');
+  assert.strictEqual(this.player.error().type, 'PLAYER_ERR_TIMEOUT');
+});
+
+QUnit.test('background timeout can be set/get via player.errors.backgroundTimeout()', function(assert) {
+  let errors = 0;
+
+  assert.strictEqual(this.player.errors.backgroundTimeout(), 300000, 'backgroundTimeout() returns default value');
+
+  this.player.errors.backgroundTimeout(20 * 1000);
+
+  assert.strictEqual(this.player.errors.backgroundTimeout(), 20000, 'backgroundTimeout() returns new value');
+
+  this.player.on('error', function() {
+    errors++;
+  });
+  this.player.src(sources);
+
+  this.player.trigger('play');
+
+  this.clock.tick(1 * 1000);
+
+  assert.ok(
+    this.player.hasClass('vjs-waiting'),
+    'the plugin adds spinner class to the player after 1 sec of no progress'
+  );
+
+  // document becomes hidden
+  document.visibilityState = 'hidden';
+  Events.trigger(document, 'visibilitychange');
+
+  this.clock.tick(19 * 1000);
+
+  assert.strictEqual(errors, 0, 'did not emit an error in background after 19 sec');
+
+  this.clock.tick(1 * 1000);
+
+  assert.strictEqual(errors, 1, 'emitted an error after 20 sec');
+  assert.strictEqual(this.player.error().code, -2, 'error code is -2');
+  assert.strictEqual(this.player.error().type, 'PLAYER_ERR_TIMEOUT');
+});
+
+QUnit.test('timeout is disabled in background if backgroundTimeout option === Infinity', function(assert) {
+  let errors = 0;
+
+  // Init with custom option
+  this.player.errors({backgroundTimeout: Infinity});
+
+  this.player.on('error', function() {
+    errors++;
+  });
+  this.player.src(sources);
+
+  this.player.trigger('play');
+
+  this.clock.tick(1 * 1000);
+
+  assert.ok(
+    this.player.hasClass('vjs-waiting'),
+    'the plugin adds spinner class to the player after 1 sec of no progress'
+  );
+
+  // document becomes hidden
+  document.visibilityState = 'hidden';
+  Events.trigger(document, 'visibilitychange');
+
+  this.clock.tick(300 * 1000);
+
+  assert.strictEqual(errors, 0, 'did not emit an error in background after 5 minutes');
+
+  this.clock.tick(300 * 1000);
+
+  assert.strictEqual(errors, 0, 'still did not emit an error in background after another 5 minutes');
+});
+
+QUnit.test('timeout is disabled in background if the player is muted', function(assert) {
```

Comment:
on a related note, if the video becomes unmuted while in the background, should we start a timeout timer?

Reply:
Yeah, I think that makes sense. We can always add it in later if we see that happening.